core/src/main/java/de/danoeh/antennapod/core/cast/CastUtils.java

```diff
@@ -0,0 +1,317 @@
+    public static final String KEY_FEED_URL = "de.danoeh.antennapod.core.cast.FeedUrl";
+    public static final String KEY_FEED_WEBSITE = "de.danoeh.antennapod.core.cast.FeedWebsite";
+    public static final String KEY_EPISODE_NOTES = "de.danoeh.antennapod.core.cast.EpisodeNotes";
+    public static final int EPISODE_NOTES_MAX_LENGTH = Integer.MAX_VALUE;
+
+    /**
+     * The field <code>AntennaPod.FormatVersion</code> specifies which version of MediaMetaData
+     * fields we're using. Future implementations should try to be backwards compatible with earlier
+     * versions, and earlier versions should be forward compatible until the version indicated by
+     * <code>MAX_VERSION_FORWARD_COMPATIBILITY</code>. If an update makes the format unreadable for
+     * an earlier version, then its version number should be greater than the
+     * <code>MAX_VERSION_FORWARD_COMPATIBILITY</code> value set on the earlier one, so that it
+     * doesn't try to parse the object.
+     */
+    public static final String KEY_FORMAT_VERSION = "de.danoeh.antennapod.core.cast.FormatVersion";
+    public static final int FORMAT_VERSION_VALUE = 1;
+    public static final int MAX_VERSION_FORWARD_COMPATIBILITY = 9999;
+
+    public static boolean isCastable(Playable media){
+        if (media == null || media instanceof ExternalMedia) {
+            return false;
+        }
+        if (media instanceof FeedMedia || media instanceof RemoteMedia){
+            String url = media.getStreamUrl();
+            if(url == null || url.isEmpty()){
+                return false;
+            }
+            switch (media.getMediaType()) {
+                case UNKNOWN:
+                    return false;
+                case AUDIO:
+                    return CastManager.getInstance().hasCapability(CastDevice.CAPABILITY_AUDIO_OUT, true);
+                case VIDEO:
+                    return CastManager.getInstance().hasCapability(CastDevice.CAPABILITY_VIDEO_OUT, true);
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Converts {@link FeedMedia} objects into a format suitable for sending to a Cast Device.
+     * Before using this method, one should make sure {@link #isCastable(Playable)} returns
+     * {@code true}.
+     *
+     * Unless media.{@link FeedMedia#loadMetadata() loadMetadata()} has already been called,
+     * this method should not run on the main thread.
+     *
+     * @param media The {@link FeedMedia} object to be converted.
+     * @return {@link MediaInfo} object in a format proper for casting.
+     */
+    public static MediaInfo convertFromFeedMedia(FeedMedia media){
+        if(media == null) {
+            return null;
+        }
+        MediaMetadata metadata = new MediaMetadata(MediaMetadata.MEDIA_TYPE_GENERIC);
+        try{
+            media.loadMetadata();
+        } catch (Playable.PlayableException e) {
+            Log.e(TAG, "Unable to load FeedMedia metadata", e);
```

Comment:
Absolutely.
1. We only need the feedItem to supply us with extra info, like feed, etc, but we can do fine just with the download url, which does not depend on it.
2. Right now, such kind of exceptions are only thrown by `ExternalMedia`, and we're not touching that at this point. In a way, there's no way that catch is ever going to trigger.